doc/developer/platform/reconciliation.md

```diff
@@ -0,0 +1,64 @@
+# Reconciliation of the ADAPTER/COMPUTE communication
+
+The current proposal for Materialize's platform includes the separation of the ADAPTER and COMPUTE components into distinct processes.
+Each represents its own failure domain and hence we need to design the communication protocols around failures and active replication (AR).
+In general, we need to define how commands and responses can be reconciled.
+
+## Command reconciliation
+
+Commands represent instructions the COMPUTE controller sends to COMPUTE instances.
+* The COMPUTE controller may restart independently of a COMPUTE instance.
+* Reconciling commands is the mechanism to match a COMPUTE instance's state to what the COMPUTE controller expects after startup.
+* On startup, the COMPUTE controller provides the current configuration to the COMPUTE instances.
```

Comment:
Should `CreateInstance` perhaps be augmented with the "initial snapshot" to remove ambiguity about whether valuable commands are coming down the pipe still? I think it shouldn't be hard to do now, with the `ComputeCommandHistory` type, which "normalizes" commands to send to a "fresh" instance.

Reply:
I pondered this a bit, and while I think we do want it, we cannot do it yet because of how the `bootstrap` method works (i.e. without a clear indication that the bootstrap is complete). Rather than try and fix that, we could punt on this until the compute controller holds the durable state, at which point it is clearer how to provide a command snapshot on restart.